libraries/AP_OpticalFlow/AP_OpticalFlow_Linux.cpp

```diff
@@ -47,50 +49,42 @@ void AP_OpticalFlow_Linux::init(void)
         return;
     }
 
-    // get pointer to i2c bus semaphore
-    AP_HAL::Semaphore *i2c_sem = hal.i2c->get_semaphore();
-    if (i2c_sem == NULL) {
-        return;
-    }
+    AP_HAL::Semaphore *sem = _dev->get_semaphore();
 
     // take i2c bus sempahore
-    if (!i2c_sem->take(200)) {
+    if (!sem->take(200)) {
         return;
     }
 
     // read from flow sensor to ensure it is not a ll40ls Lidar (which can be on 0x42)
     // read I2C_FRAME_SIZE bytes, the ll40ls will error whereas the flow happily returns data
     uint8_t val[I2C_FRAME_SIZE];
-    if (hal.i2c->readRegisters(PX4FLOW_I2C_ADDRESS, 0, I2C_FRAME_SIZE, val) != 0) {
-        i2c_sem->give();
+    if (!_dev->read_registers(0, val, I2C_FRAME_SIZE)) {
+        sem->give();
         return;
     }
 
     // success
     initialised = true;
-    i2c_sem->give();
+    sem->give();
 }
 
 bool AP_OpticalFlow_Linux::request_measurement()
 {
     // send measure request to sensor
-    uint8_t cmd = PX4FLOW_REG;
-    if (hal.i2c->writeRegisters(PX4FLOW_I2C_ADDRESS, cmd, 0, nullptr) != 0) {
+    uint8_t cmd = HAL_OPTFLOW_PX4FLOW_REGISTER;
+    if (!_dev->write_register(cmd, 0)) {
         return false;
     }
     return true;
 }
 
 bool AP_OpticalFlow_Linux::read(optical_flow_s* report)
 {
-    // get pointer to i2c bus semaphore
-    AP_HAL::Semaphore *i2c_sem = hal.i2c->get_semaphore();
-    if (i2c_sem == NULL) {
-        return false;
-    }
+    AP_HAL::Semaphore *sem = _dev->get_semaphore();
```

Comment:
same as above... there are very few places using it, so just call _dev->get_semaphore() in each place